src/api/TPLink.ts

```diff
@@ -86,6 +86,9 @@ export default class TPLink {
     command: T,
     ...args: Parameters<Commands[T]>
   ): Promise<CommandReturnType<T>> {
+    if (command === 'power' && this._prevPowerState == args[0]) {
```

Comment:
For the second check in if, use `===`

Reply:
Also, move the login inside the lock.acquire, because they won't be executed in the same time, and it safer